### ISSUE TITLE
[MINSTALL-138] option to calculate more checksum such sha-256 sha-512

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-install-plugin#300]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** opened **[MINSTALL-138](https://issues.apache.org/jira/browse/MINSTALL-138?redirect=false)** and commented

currently install generate only sha-1 we should be able to generate sha-256 and sha-512 as well.
NOTE: sha-512 will be required by Apache policy.

---

**Affects:** 2.5.2

**Issue Links:**
- [MDEPLOY-231](https://issues.apache.org/jira/browse/MDEPLOY-231) Move checksum generation from install to deploy plugin
 (_**"is blocked by"**_)
- [MINSTALL-143](https://issues.apache.org/jira/browse/MINSTALL-143) Move checksum generation from install to deploy plugin
 (_**"is blocked by"**_)
- [MINSTALL-82](https://issues.apache.org/jira/browse/MINSTALL-82) Add sha 256 support
 (_**"contains"**_)
- [MSHARED-704](https://issues.apache.org/jira/browse/MSHARED-704) Option to create SHA256/SHA512
 (_**"is part of"**_)
- [MRESOLVER-140](https://issues.apache.org/jira/browse/MRESOLVER-140) Default to SHA-1 and MD5 hashing algorithms

- [MINSTALL-134](https://issues.apache.org/jira/browse/MINSTALL-134) Remove checksum generation

- [MPOM-205](https://issues.apache.org/jira/browse/MPOM-205) create SHA-512 checksum for source-release archive(s) in target/checkout/target/ during release

- [MPOM-244](https://issues.apache.org/jira/browse/MPOM-244) Deploy SHA-512 for source-release to Remote Repository

1 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

I'm not convinced install or deploy are the right place to create checksums required by Apache (source) release distribution policy http://www.apache.org/dev/release-distribution

Maven repository checksums are applied to every file deployed, in general .pom, .jar, -sources.jar and -javadoc.jar (see https://repo.maven.apache.org/maven2/org/apache/maven/maven-artifact/3.5.4/ for example)
Apache source release distribution checksum is only for -source-release.zip

now that MPOM-205 works, I'm convinced that creating new checksums for Apache release distribution policy at Maven repository level is not the right approach

if we want the source release zip sha-512 checksum, we can attach the artifact specifically for this source-release zip

but adding sha-512 checksums for absolutely every file in the Maven repository is just overkill IMHO

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

Nexus now supports SHA-256/512 checksums (https://issues.sonatype.org/browse/NEXUS-21802) and also the maven-resolver will support those with the upcoming version (MRESOLVER-56). But as Herve outlined, it should be configurable. With the move from m-install-plugin to m-deploy-plugin in MINSTALL-143 the parameter `createChecksum` has been removed without any replacement. So currently there is no option to configure the generation of checksums.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

> but adding sha-512 checksums for absolutely every file in the Maven repository is just overkill IMHO

I don't agree here. We are talking about 128 byte for SHA512. IMHO always adding both SHA1 and MD5 is no longer necessary. Maybe just using SHA512 instead of SHA1 (=40 bytes) in addition to MD5 (for older clients) is a reasonable default. That way we are only talking about 88 bytes more for checksums.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

Now that we have MRESOLVER-56 we can probably close this one as won't fix.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I consider this one as wontfix and [Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin) has layed out.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Please raise if you still consider this to be an issue.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) As the default has been changed in MRESOLVER-140 we need a way to configure which checksum to generate with plugin parameters. Can you please reopen?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

No, as mentioned numerous times, install does **not** require any checksums. In fact, this is the wrong place. Transport requires and MDEPLOY delegates this Resolver. So Resolver must take care of this. It be a violation of separation of concerns when this would be configurable here.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

You are looking for something completely else: https://github.com/apache/maven/blob/master/apache-maven/pom.xml#L310-L336

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

This would lead to generating SHA1 and MD5 for the generated checksum artifacts (if you deploy those as well). If you don't deploy SHA-512 (like you do in https://github.com/apache/maven/blob/master/apache-maven/pom.xml#L310-L336) the validation of the checksum cannot easily be automated (as it cannot be retrieved from the staging repository).
